### DISCUSSION TITLE
sx126x rx timeout

### DISCUSSION THREAD

**Discussion by macleayislandbruce** [jgromes/RadioLib#387] [Q&A]
Hi, first congratulations to Jan for the library, I can see an enormous amount of work went into it, great examples too.

I am using SX1262 LoRa radios, basically built on the ping/pond interrupt example. However I cannot get the rx timeout to work. From the class constructor it should be radio.startReceive(timeout in multiples of 15.625 microsecs), I have tried to set timeout to 200mS, ie 15.625uS x 12800 = 200,000uS so we get "uint32_t rxTimeOut = `12800; radio.startReceive(rxTimeOut);"? this doesn't work, timeout is still default of infinite. What is the format for setting the timeout? Sorry if this is an obvious newby mistake.
Regard Bruce

**Comment by jgromes** [comment]
> I am using SX1262 LoRa radios, basically built on the ping/pond interrupt example

In the interrupt mode, RX timeout is not used, the radio is placed to countinuous reception mode and never times out. The timeout only matters for blocking reception.

**Reply by macleayislandbruce** [reply]
Ok thanks Jan. I made an error in my code, it was not doing what I thought it was, all good now. However as I understand it startReceive() is only used for interupt mode, receive() is blocking mode. In the class reference startReceive() has timeout options, receive() has length (of data) options, so timeout does what??
I assume (haven't tried yet) if radio is in continious receive mode, I can terminate this at any time using startTransmit(), reset txFlag and enableInterrupt?

**Reply by jgromes** [reply]
`startReceive()` is actually used internally when calling `receive()`, since it does all the configuration that must be done to enable reception. The timeout parameter exists to facilitate this.

> I can terminate this at any time using startTransmit()

The receive interrupt example lists all the methods that will stop continuous reception:

```c++
  // if needed, 'listen' mode can be disabled by calling
  // any of the following methods:
  //
  // radio.standby()
  // radio.sleep()
  // radio.transmit();
  // radio.receive();
  // radio.readData();
  // radio.scanChannel();
```

You probably want to use `radio.standby()`.

> reset txFlag and enableInterrupt

In the context of the pin pong example, `enableInterrupt` won't achieve that. `transmitFlag` might, but you'd have to be careful when you modify it, I haven't tried that.

**Reply by macleayislandbruce** [reply]
Thanks Jan, I am now using a receive timer and then an if loop to use radio.standby and then radio.startTransmit. This works perfectly, Originating radio, if it doen't receive a response, after timeout, it transmits another packet then listens again etc.

Many thanks for your help and thanks again for a great library!
Regards Bruce
